### PR TITLE
Fixing boxplot for duckdb native

### LINKED ISSUE #728: support histogram when using a DBAPI connection
when using a generic DBAPI connection, I get an error when trying to use boxplot:

```
In [1]: %load_ext sql

In [2]: import duckdb; conn = duckdb.connect()

In [3]: %sql conn

In [6]: %sql select * from iris.csv
Running query in '<duckdb.DuckDBPyConnection object at 0x11e1921f0>'
Out[6]:
+--------------+-------------+--------------+-------------+-------------+
| sepal length | sepal width | petal length | petal width |    class    |
+--------------+-------------+--------------+-------------+-------------+
|     5.1      |     3.5     |     1.4      |     0.2     | Iris-setosa |
|     4.9      |     3.0     |     1.4      |     0.2     | Iris-setosa |
|     4.7      |     3.2     |     1.3      |     0.2     | Iris-setosa |
|     4.6      |     3.1     |     1.5      |     0.2     | Iris-setosa |
|     5.0      |     3.6     |     1.4      |     0.2     | Iris-setosa |
|     5.4      |     3.9     |     1.7      |     0.4     | Iris-setosa |
|     4.6      |     3.4     |     1.4      |     0.3     | Iris-setosa |
|     5.0      |     3.4     |     1.5      |     0.2     | Iris-setosa |
|     4.4      |     2.9     |     1.4      |     0.2     | Iris-setosa |
|     4.9      |     3.1     |     1.5      |     0.1     | Iris-setosa |
+--------------+-------------+--------------+-------------+-------------+

In [7]: %sqlplot boxplot --table iris.csv --column class
Installed osx event loop hook.
RuntimeError: boxplot is not supported for a custom engine
If you need help solving this issue, send us a message: https://ploomber.io/community

```

the error is raised here: https://github.com/ploomber/jupysql/blob/aa360f9505c7b0e3ad8968cf38b2ccfc9df9d4a1/src/sql/plot.py#L145

and it suggests that we are not allowing it because some databases do not support the percentile disc function; however this doesn't have to do with being a dbapi connection or a sqlalchemy one. they'll both send the query to the db and the query will work just fine if it has the percentile_disc function

is there are other reason why we're not supporting it? @yafimvo

### PR DESCRIPTION
**Pull request by mehtamohit013**
## Describe your changes
Add support for duckdb native connection for boxplot

## Issue number

Closes #728 

## Checklist before requesting a review

- [X] Performed a self-review of my code
- [X] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [X] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [X] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--762.org.readthedocs.build/en/762/

### CODE REVIEW

**Comment by mehtamohit013** [comment]
> but we need to ensure that the boxplot is actually displayed and it produces the some one as it would with duckdb + sqlalchemy

The same image is working, I just copy the image for convenience. 

> we also need to update the docs because the duckdb tutorial that uses native connections isn't showing the boxplot example

Can you give me more details about it? Which tutorial are you specifically talking about here?
@edublancas

**Comment by mehtamohit013** [comment]
I don't quite understand how questDB testing is working.  I added `psycopg2` in `INTEGRATION` in `setup.py`, but the tests are still failing: `ModuleNotFoundError: No module named 'pyscopg2'`.

Any thoughts on this @edublancas ?

**Comment by edublancas** [comment]
> I don't quite understand how questDB testing is working. I added `psycopg2` in `INTEGRATION` in `setup.py`, but the tests are still failing: `ModuleNotFoundError: No module named 'pyscopg2'`.
> 
> Any thoughts on this @edublancas ?

these database drivers always give trouble 😪

we already have psycopg2 as a dependency (psycopg2-binary) so I'm surprised this is failing, what happens if you remove the line you added? master is passing so questdb tests should pass without modifications. maybe it's something else that's causing the issue?

**Comment by edublancas** [comment]
wait, master is failing due it's due to a different thing. I'm fixing it now

File: doc/tutorials/duckdb-native-sqlalchemy.md [hunk 1]
```diff
@@ -128,6 +128,7 @@ To suppress it, add this at the top of your notebook/script:
 ```{code-cell} ipython3
 from sql.warnings import JupySQLDataFramePerformanceWarning
 import warnings
+
```
Comment (neelasha23) [inline doc/tutorials/duckdb-native-sqlalchemy.md:131]: Based on this [comment](https://github.com/ploomber/jupysql/issues/728#issuecomment-1652351592) it looks like we need to add a boxplot example in this guide

File: src/tests/integration/test_questDB.py [hunk 2]
```diff
@@ -546,21 +547,21 @@ def test_sqlcmd_not_supported_error(ip_questdb, query, capsys):
...
-            f"boxplot {NOT_SUPPORTED_SUFFIX}",
+            "unbalanced",
         ),
     ],
 )
 def test_ggplot_boxplot_not_supported_error(
     ip_questdb, penguins_no_nulls_questdb, penguins_data, func, expected_error_message
 ):
-    with pytest.raises(UsageError) as err:
+    with pytest.raises(pg2.DatabaseError) as err:
```
Comment (yafimvo) [inline src/tests/integration/test_questDB.py:564]: I'm not sure what this test does. are we testing boxplot with unbalanced data? if so, let's change the name of the test. also, shouldn't it raise `UsageError`?

**Review by yafimvo — CHANGES_REQUESTED** [review]
Please rebase

**Review by edublancas — CHANGES_REQUESTED** [review]
address comments from existing reviews

**Comment by mehtamohit013** [comment]
> address comments from existing reviews

I'm sorry for the delay. I will definitely complete it before tomorrow morning(EST)

File: src/tests/integration/test_questDB.py (hunk 2 above)
Comment (mehtamohit013) [reply src/tests/integration/test_questDB.py]: Removed the test, as discussed in monday meeting

**Review by edublancas — APPROVED** [review]

**Pull request merged**